File: example/src/main/java/ru/virgil/example/image/PrivateImageFile.java
package ru.virgil.example.image;

import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import ru.virgil.example.system.IdentifiedEntity;
import ru.virgil.example.user.UserDetails;
import ru.virgil.utils.image.IPrivateImage;

import javax.persistence.ManyToOne;
import java.nio.file.Path;

@javax.persistence.Entity
@EqualsAndHashCode(callSuper = true)
@NoArgsConstructor
@Data
public class PrivateImageFile extends IdentifiedEntity implements IPrivateImage<UserDetails> {

    @ManyToOne
    private UserDetails owner;
    private String fileLocation;

    @Override
    public Path getFileLocation() {
        return Path.of(fileLocation);
    }

    @Override
    public void setFileLocation(Path location) {
        this.fileLocation = location.toString();
    }

}

File: example/src/main/java/ru/virgil/example/image/PrivateImageFile.kt
package ru.virgil.example.image

import ru.virgil.example.system.entity.OwnedEntity
import ru.virgil.example.user.UserDetails
import ru.virgil.utils.image.PrivateImage
import java.nio.file.Path
import javax.persistence.Entity
import javax.persistence.Transient

@Entity
class PrivateImageFile(
    owner: UserDetails,
    fileLocation: Path,
) : OwnedEntity(owner), PrivateImage<UserDetails> {

    @get:Transient
    override var fileLocation: Path
        get() = Path.of(location)
        set(value) {
            location = value.toString()
        }

    private var location: String = fileLocation.toString()
}
